### PR TITLE
Start changelog for 7.8.0

### PR DESCRIPTION
**Pull request by mikemadden42**

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -3,6 +3,574 @@
...
+- Add static mapping for metricsets under aws module. {pull}17614[17614] {pull}17650[17650]
+- Add dashboard for googlecloud storage metricset. {pull}18172[18172]
+- Collect new `bulk` indexing metrics from Elasticsearch when `xpack.enabled:true` is set. {issue} {pull}17992[17992]
+- Remove requirement to connect as sysdba in Oracle module {issue}15846[15846] {pull}18182[18182]
+- Update MSSQL module to fix some SSPI authentication and add brackets to USE statements {pull}17862[17862]]
+
+*Winlogbeat*
+
+- Set process.command_line and process.parent.command_line from Sysmon Event ID 1. {pull}17327[17327]
+- Add support for event IDs 4673,4674,4697,4698,4699,4700,4701,4702,4768,4769,4770,4771,4776,4778,4779,4964 to the Security module {pull}17517[17517]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:119]:
```suggestion
- Add support for event IDs 4673,4674,4697,4698,4699,4700,4701,4702,4768,4769,4770,4771,4776,4778,4779,4964 to the Security module. {pull}17517[17517]
```

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -3,6 +3,574 @@
...
+- Add registry and code signature information and ECS categorization fields for sysmon module {pull}18058[18058]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:120]:
```suggestion
- Add registry and code signature information and ECS categorization fields for sysmon module. {pull}18058[18058]
```

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -3,6 +3,574 @@
...
+==== Breaking changes
+
+*Affecting all Beats*
+
+- Introduce APM libbeat instrumentation, active when running the beat with ELASTIC_APM_ACTIVE=true. {pull}17938[17938]
+
+*Filebeat*
+
+- Improve ECS field mappings in panw module.  event.outcome now only contains success/failure per ECS specification. {issue}16025[16025] {pull}17910[17910]
+- Improve ECS categorization field mappings for nginx module. http.request.referrer is now lowercase & http.request.referrer only populated when nginx sets a value {issue}16174[16174] {pull}17844[17844]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:19]:
```suggestion
- Improve ECS categorization field mappings for nginx module. `http.request.referrer` is now lowercase, and it is only populated when nginx sets a value. {issue}16174[16174] {pull}17844[17844]
```

File: CHANGELOG.asciidoc [hunk 4]
```diff
@@ -3,6 +3,574 @@
...
 
+[[release-notes-7.8.0]]
+=== Beats version 7.8.0
+https://github.com/elastic/beats/compare/v7.7.0...v7.8.0[View commits]
+
+==== Breaking changes
+
+*Affecting all Beats*
+
+- Introduce APM libbeat instrumentation, active when running the beat with ELASTIC_APM_ACTIVE=true. {pull}17938[17938]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:14]:
```suggestion
- Introduce APM instrumentation, which is active when running the beat with `ELASTIC_APM_ACTIVE=true`. {pull}17938[17938]
```

File: CHANGELOG.asciidoc [hunk 5]
```diff
@@ -3,6 +3,574 @@
...
+
+*Filebeat*
+
+- Improve ECS field mappings in panw module.  event.outcome now only contains success/failure per ECS specification. {issue}16025[16025] {pull}17910[17910]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:18]:
```suggestion
- Improve ECS field mappings in panw module.  `event.outcome` now only contains success or failure, as recommended by the {ecs-ref}/ecs-event.html[ECS specification]. {issue}16025[16025] {pull}17910[17910]
```

File: CHANGELOG.asciidoc [hunk 6]
```diff
@@ -3,6 +3,574 @@
...
+- Improve ECS categorization field mappings for nginx module. http.request.referrer is now lowercase & http.request.referrer only populated when nginx sets a value {issue}16174[16174] {pull}17844[17844]
+- Improve ECS field mappings in santa module. move hash.sha256 to process.hash.sha256 & move certificate fields to santa.certificate . {issue}16180[16180] {pull}17982[17982]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:20]:
```suggestion
- Improve ECS field mappings in santa module. `hash.sha256` is moved to `process.hash.sha256`, and certificate fields are now under `santa.certificate`. {issue}16180[16180] {pull}17982[17982]
```

File: CHANGELOG.asciidoc [hunk 7]
```diff
@@ -3,6 +3,574 @@
...
+- Add system module login dataset ECS categorization fields. {pull}18034[18034]
+- Add system module package dataset ECS categorization fields. {pull}18033[18033]
+- Add ECS categories for system module host dataset. {pull}18031[18031]
+- Add system module socket dataset ECS categorization fields. {pull}18036[18036]
+- Add file integrity module ECS categorization fields. {pull}18012[18012]
+- Add `file.mime_type`, `file.extension`, and `file.drive_letter` for file integrity module. {pull}18012[18012]
+
+*Filebeat*
+
+- Add source field in k8s events {pull}17209[17209]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:69]:
```suggestion
- Add source field in k8s events. {pull}17209[17209]
```

File: CHANGELOG.asciidoc [hunk 8]
```diff
@@ -3,6 +3,574 @@
...
+- Add Metricbeat IIS module dashboards. {pull}17966[17966]
+- Add dashboard for the azure database account metricset. {pull}17901[17901]
+- Allow partial region and zone name in googlecloud module config. {pull}17913[17913]
+- Add aggregation aligner as a config parameter for googlecloud stackdriver metricset. {issue}17141[[17141] {pull}17719[17719]
+- Move the perfmon metricset to GA. {issue}16608[16608] {pull}17879[17879]
+- Stack Monitoring modules now auto-configure required metricsets when `xpack.enabled: true` is set. {issue}16471[[16471] {pull}17609[17609]
+- Add static mapping for metricsets under aws module. {pull}17614[17614] {pull}17650[17650]
+- Add dashboard for googlecloud storage metricset. {pull}18172[18172]
+- Collect new `bulk` indexing metrics from Elasticsearch when `xpack.enabled:true` is set. {issue} {pull}17992[17992]
+- Remove requirement to connect as sysdba in Oracle module {issue}15846[15846] {pull}18182[18182]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:113]:
```suggestion
- Remove requirement to connect as sysdba in Oracle module. {issue}15846[15846] {pull}18182[18182]
```

File: CHANGELOG.asciidoc (hunk 4 above)
Comment (dedemorton) [reply CHANGELOG.asciidoc:14]: Can we drop "libbeat" from this description? Just thinking that libbeat is an implementation detail that end users don't really need to know about.

File: CHANGELOG.asciidoc [hunk 9]
```diff
@@ -3,6 +3,574 @@
...
+- Further revise check for bad data in docker/memory. {pull}17400[17400]
+- Add a switch to the driver definition on SQL module to use pretty names {pull}17378[17378]
+
+==== Added
+
+*Affecting all Beats*
+
+- Update supported versions of `redis` output. {pull}17198[17198]
+- Add `replace` processor for replacing string values of fields. {pull}17342[17342]
+- Add `urldecode` processor to for decoding URL-encoded fields. {pull}17505[17505]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:44]:
```suggestion
- Add `urldecode` processor for decoding URL-encoded fields. {pull}17505[17505]
```

File: CHANGELOG.asciidoc [hunk 10]
```diff
@@ -3,6 +3,574 @@
...
+- Stack Monitoring modules now auto-configure required metricsets when `xpack.enabled: true` is set. {issue}16471[[16471] {pull}17609[17609]
+- Add Metricbeat IIS module dashboards. {pull}17966[17966]
+- Add dashboard for the azure database account metricset. {pull}17901[17901]
+- Allow partial region and zone name in googlecloud module config. {pull}17913[17913]
+- Add aggregation aligner as a config parameter for googlecloud stackdriver metricset. {issue}17141[[17141] {pull}17719[17719]
+- Move the perfmon metricset to GA. {issue}16608[16608] {pull}17879[17879]
+- Stack Monitoring modules now auto-configure required metricsets when `xpack.enabled: true` is set. {issue}16471[[16471] {pull}17609[17609]
+- Add static mapping for metricsets under aws module. {pull}17614[17614] {pull}17650[17650]
+- Add dashboard for googlecloud storage metricset. {pull}18172[18172]
+- Collect new `bulk` indexing metrics from Elasticsearch when `xpack.enabled:true` is set. {issue} {pull}17992[17992]
```
[the remark this answers is not in the document]
Comment (mikemadden42) [reply CHANGELOG.asciidoc:112]: @ycombinator Can you let me know the issue number, and I'll add it.

[the remark this answers is not in the document]
Comment (ycombinator) [reply CHANGELOG.asciidoc:112]: Oops, sorry, it should be 17977.

File: CHANGELOG.asciidoc [hunk 11]
```diff
@@ -3,6 +3,574 @@
...
+- Fix goroutine leak and Elasticsearch output file descriptor leak when output reloading is in use. {issue}10491[10491] {pull}17381[17381]
+- Fix panic when assigning a key to a `nil` value in an event. {pull}18143[18143]
+
+*Heartbeat*
+
+- Fixed TCP TLS checks to properly validate hostnames, this broke in 7.x and only worked for IP SANs. {pull}17549[17549]
+
+*Metricbeat*
+
+- Further revise check for bad data in docker/memory. {pull}17400[17400]
```
[the remark this answers is not in the document]
Comment (mikemadden42) [reply CHANGELOG.asciidoc]: I changed the wording for this entry.

File: CHANGELOG.asciidoc [hunk 12]
```diff
@@ -3,6 +3,574 @@
...
+- Make `decode_cef` processor GA. {pull}17944[17944]
+- Added new Fortigate Syslog filebeat module. {pull}17890[17890]
+- Improve ECS categorization field mappings in redis module. {issue}16179[16179] {pull}17918[17918]
+- Improve ECS categorization field mappings in rabbitmq module. {issue}16178[16178] {pull}17916[17916]
+- Improve ECS categorization field mappings in postgresql module. {issue}16177[16177] {pull}17914[17914]
+- Improve ECS categorization field mappings for nginx module. {issue}16174[16174] {pull}17844[17844]
+- Add support for Google Application Default Credentials to the Google Pub/Sub input and Google Cloud modules. {pull}15668[15668]
+- Improve ECS categorization field mappings for zeek module. {issue}16029[16029] {pull}17738[17738]
+- Improve ECS categorization field mappings for netflow module. {issue}16135[16135] {pull}18108[18108]
+- Added an input option `publisher_pipeline.disable_host` to disable `host.name`
```
[the remark this answers is not in the document]
Comment (mikemadden42) [reply CHANGELOG.asciidoc]: @andrewkroh Can you let me know the issue & PR for this line, and I'll add it to the changelog.

- Added an input option `publisher_pipeline.disable_host` to disable `host.name`

It's likely related to this commit:

```
commit 84078fbcb72baa361d87ae43884140b5401927df
Author: Andrew Kroh <andrew.kroh@elastic.co>
Date:   Tue May 5 13:19:43 2020 -0400

    Add a disable_host option to Filebeat inputs (#18159) (#18188)
    
    This adds a configuration option `publisher_pipeline.disable_host` to disable the addition
    of `host.name` in events. By default Filebeat adds `host.name` to all events and we want
    to be able to disable this for data sources that do not originate on the host (like cloud logs).
    
    Relates #13920
    
    (cherry picked from commit 6ea21a9c4b433ce4dff83d815f856fce17d16f4e)
```

File: CHANGELOG.asciidoc [hunk 13]
```diff
@@ -3,6 +3,574 @@
...
+
+- Refactor windows/perfmon metricset configuration options and event output. {pull}17596[17596]
+- Add more detailed error messages, system tests and small refactoring to the service metricset in windows. {pull}17725[17725]
+- Stack Monitoring modules now auto-configure required metricsets when `xpack.enabled: true` is set. {issue}16471[16471] {pull}17609[17609]
+- Add Metricbeat IIS module dashboards. {pull}17966[17966]
+- Add dashboard for the azure database account metricset. {pull}17901[17901]
+- Allow partial region and zone name in googlecloud module config. {pull}17913[17913]
+- Add aggregation aligner as a config parameter for googlecloud stackdriver metricset. {issue}17141[17141] {pull}17719[17719]
+- Move the perfmon metricset to GA. {issue}16608[16608] {pull}17879[17879]
+- Stack Monitoring modules now auto-configure required metricsets when `xpack.enabled: true` is set. {issue}16471[[16471] {pull}17609[17609]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:109]:
```suggestion
- Stack Monitoring modules now auto-configure required metricsets when `xpack.enabled: true` is set. {issue}16471[16471] {pull}17609[17609]
```

File: CHANGELOG.asciidoc [hunk 14]
```diff
@@ -3,6 +3,574 @@
...
+- Fix panic when assigning a key to a `nil` value in an event. {pull}18143[18143]
+
+*Heartbeat*
+
+- Fix TCP TLS checks to properly validate hostnames. In previous 7.x versions, this only worked for IP SANs. {pull}17549[17549]
+
+*Metricbeat*
+
+- No longer send NaNs for memory metrics that don't exist on the platform being monitored. {pull}17400[17400]
+- Add a switch to the driver definition on SQL module to use pretty names {pull}17378[17378]
```
Comment (dedemorton) [inline CHANGELOG.asciidoc:36]:
```suggestion
- Add a switch to the driver definition on SQL module to use pretty names. {pull}17378[17378]
```

File: CHANGELOG.asciidoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (mikemadden42) [reply CHANGELOG.asciidoc]: I've fixed the missing links. 😄

File: CHANGELOG.asciidoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (mikemadden42) [reply CHANGELOG.asciidoc]: I've fixed this.

**Pull request merged**